internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -977,8 +1004,11 @@ func flattenAdvancedReplicationSpec(ctx context.Context, apiObject *admin.Replic
 	}
 
 	tfMap := map[string]any{}
-	tfMap["id"] = apiObject.GetId()
```

Comment:
when using new api we explicitly don't set `id` attribute so it is clear we are relying on `external_id`.